[mlir] Rename mlir-cpu-runner to mlir-runner

andfau-amd opened the pull request:
With the removal of mlir-vulkan-runner (as part of #73457) in e7e3c45bc70904e24e2b3221ac8521e67eb84668, mlir-cpu-runner is now the only runner for all CPU and GPU targets, and the "cpu" name has been misleading for some time already. This commit renames it to mlir-runner.

kuhar commented on mlir/tools/mlir-runner/mlir-runner.cpp:
```diff
@@ -1,4 +1,4 @@
-//===- mlir-cpu-runner.cpp - MLIR CPU Execution Driver---------------------===//
+//===- mlir-runner.cpp - MLIR CPU Execution Driver-------------------------===//
```
```suggestion
//===- mlir-runner.cpp - MLIR CPU Execution Driver ------------------------===//
```

kuhar reviewed (approved):
LGTM but please wait for at least a couple more approvals from other code owners

kuhar commented:
Should we send an PSA on discourse about this? @Hardcode84 suggested this on discord.

andfau-amd commented:
Hi, I created a Discourse thread here: https://discourse.llvm.org/t/psa-renaming-of-mlir-cpu-runner-to-mlir-runner/84201.

I hope that's roughly what you had in mind, I am not really familiar with the Discourse norms.

banach-space commented on mlir/test/CMakeLists.txt:
```diff
@@ -22,8 +22,8 @@ if (MLIR_INCLUDE_INTEGRATION_TESTS)
       "If arch-specific Arm integration tests run emulated, pass these as parameters to the emulator.")
-  set(ARM_EMULATOR_MLIR_CPU_RUNNER_EXECUTABLE "" CACHE STRING
-      "If arch-specific Arm integration tests run emulated, use this Arm native mlir-cpu-runner.")
+  set(ARM_EMULATOR_MLIR_RUNNER_EXECUTABLE "" CACHE STRING
```
Thanks for flagging this, Renato!

From Arm's perspective, we don't require this since we run everything natively on AArch64. This is more relevant for those who need to cross-compile. 

I'm not aware of anyone actively using it, so the renaming should be fine. If any issues arise following this change, I'm happy to help folks adjust their setups.

rengolin replied:
Thanks! Should be fine to keep the change, then. It should be trivial to fix this merge conflict when some downstream project rebases the code.

andfau-amd committed:
[mlir] Rename mlir-cpu-runner to mlir-runner

With the removal of mlir-vulkan-runner (as part of #73457) in
e7e3c45bc70904e24e2b3221ac8521e67eb84668, mlir-cpu-runner is now the
only runner for all CPU and GPU targets, and the "cpu" name has been
misleading for some time already. This commit renames it to mlir-runner.
